xicam/gui/widgets/plotwidgetmixins.py

```diff
@@ -109,13 +110,114 @@ def showLabel(self, item, points):
         self._curvepoint.setIndex(list(item.scatter.points()).index(point))
 
 
+
+class PlotWidget(QWidget):
+    """Wrapper for pyqtgraph.PlotWidget that allows mixin extensibility.
+
+    Provides three Qt layouts:
+    1. main_layout - vertical box layout that contains the plot view and the bottom widget area
+    2. bottom_layout - vertical box layout that contains extra widgets
+    3. inner_layout - extra horizontal layout inside the bottom layout
+
+    Use the API provided by pyqtgraph.PlotWidget for plotting.
+    """
+    def __init__(self, *args, **kwargs):
+        super(QWidget, self).__init__(*args, **kwargs)
+        self.plot_widget = pg.PlotWidget()
+
+        self.main_layout = QVBoxLayout()
+        self.bottom_layout = QVBoxLayout()
+        self.inner_layout = QHBoxLayout()
+
+        self.main_layout.addWidget(self.plot_widget)
+        self.main_layout.addLayout(self.bottom_layout)
+        self.bottom_layout.addLayout(self.inner_layout)
+
+        self.setLayout(self.main_layout)
+
+    def __getattr__(self, attr):
+        # implicitly wrap methods from plotItem
+        if hasattr(self.plot_widget, attr):
+            m = getattr(self.plot_widget, attr)
+            if hasattr(m, '__call__'):
+                return m
+        raise AttributeError(attr)
+
+
+class AutoRotateCheckbox(PlotWidget):
+    # TODO: move to image view mixins
+    def __init__(self, *args, **kwargs):
+        super(AutoRotateCheckbox, self).__init__(*args, **kwargs)
+        rotate_checkbox = QCheckBox("Rotate")
+        # rotate_checkbox.stateChanged.connect(self._rotate_checkbox_clicked)
+        self.bottom_layout.addWidget(QCheckBox(rotate_checkbox))
+
+    # def _rotate_checkbox_clicked(self, checked):
+    #     if checked:
+    #         ...
+    #     else:
+    #         ...
+
+
+class OffsetPlots(PlotWidget):
+    """Create a visual offset in the plots"""
+    # TODO: implement the offset code
+    def __init__(self, *args, **kwargs):
+        super(OffsetPlots, self).__init__(*args, **kwargs)
+        self.offset_box = QDoubleSpinBox()
+        self.offset_box.setMinimum(0.0)
+        self.offset_box.setDecimals(1)
+        self.offset_box.setSingleStep(0.1)
+        self.offset_button = QPushButton("Enable Offset")
+        self.offset_button.setCheckable(True)
+        self.offset_button.toggled.connect(self._offset_toggled)
+
+        layout = QHBoxLayout()
+        layout.addWidget(self.offset_box)
+        layout.addWidget(self.offset_button)
+        self.inner_layout.addLayout(layout)
+
+    def _offset_toggled(self, enabled):
+        if enabled:
+            self.offset_button.setText("Disable Offset")
+        else:
+            self.offset_button.setText("Enable Offset")
+
+
+class LogButton(PlotWidget):
```

Comment:
Rename to "XLogButton" (?)

Reply:
log buttons have been refactored to include both x and y (you probably will want to be able to toggle log mode for both axes)